winbuild/dir.cpp

```diff
@@ -48,22 +50,104 @@ class WinDirHandle : public watchman_dir_handle {
   }
 
   const watchman_dir_ent* readDir() override {
-    if (!info_) {
-      if (!GetFileInformationByHandleEx(
-              (HANDLE)h_.handle(), FileFullDirectoryInfo, buf_, sizeof(buf_))) {
-        if (GetLastError() == ERROR_NO_MORE_FILES) {
-          return nullptr;
+    if ((getenv("WATCHMAN_WIN7_COMPAT") && 
+         getenv("WATCHMAN_WIN7_COMPAT")[0] == '1')) {
+      // Before Windows 8, the FileFullDirectoryInfo parameter is not supported when 
```

Comment:
This seems like an incomplete thought being expressed in the comment; can you fix it up, or just truncate it to "not supported"?